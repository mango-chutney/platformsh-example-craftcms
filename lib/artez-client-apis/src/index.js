/*
 * index.js
 *
 * @flow
 */

import * as regapi from './regapi';
import * as registrant from './registrant';
import * as webgetservice from './webgetservice';

export { regapi, registrant, webgetservice };

export default { regapi, registrant, webgetservice };
