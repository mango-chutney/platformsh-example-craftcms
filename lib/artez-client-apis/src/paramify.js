/*
 * paramify.js
 *
 * @flow
 */

export default function(params: Object): string {
  return Object.keys(params)
    .filter(key => params[key] !== undefined)
    .map(
      key =>
        params[key] === undefined
          ? ''
          : `${key}=${encodeURIComponent(params[key])}`,
    )
    .join('&');
}
