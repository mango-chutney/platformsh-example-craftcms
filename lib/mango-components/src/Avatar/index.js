// @flow

export { default as Avatar } from './Avatar';
export { default as AvatarComponent } from './AvatarComponent';
export { default as AvatarComposer } from './AvatarComposer';
