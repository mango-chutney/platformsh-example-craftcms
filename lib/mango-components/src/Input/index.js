// @flow

export { default as createInputIdAttribute } from './createInputIdAttribute';
export { default as createLabelForAttribute } from './createLabelForAttribute';
export { default as createLabelObject } from './createLabelObject';
export { default as createLabelProps } from './createLabelProps';
export { default as defaultTheme } from './styles';
export { default as Input } from './Input';
export { default as InputComponent } from './InputComponent';
export { default as InputComposer } from './InputComposer';
export { default as InputDecoratorComponent } from './InputDecoratorComponent';
export { default as InputLabelComponent } from './InputLabelComponent';
export {
  default as createFormControlElementProps,
} from './createFormControlElementProps';
export {
  default as createInputDecoratorProps,
} from './createInputDecoratorProps';
