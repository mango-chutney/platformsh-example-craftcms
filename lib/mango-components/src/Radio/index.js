// @flow

export { default as defaultTheme } from './styles';
export { default as Radio } from './Radio';
export { default as RadioComposer } from './RadioComposer';
export { default as RadioInputComponent } from './InputComponent';
export {
  default as RadioInputDecoratorComponent,
} from './InputDecoratorComponent';
export { default as RadioLabelComponent } from './LabelComponent';
export { default as RadioWrapperComponent } from './WrapperComponent';
