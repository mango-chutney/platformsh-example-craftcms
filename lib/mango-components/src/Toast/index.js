// @flow

export { default as Toast } from './Toast';
export { default as PersistentToast } from './PersistentToast';
export { default as ToastPortal } from './ToastPortal';
export {
  ToastConsumer,
  default as ToastProviderComposer,
  transitionStates,
} from './ToastProviderComposer';
export { default as PersistentToastComposer } from './PersistentToastComposer';
export { default as RainbowToastComponent } from './RainbowToastComponent';
export { default as ToastComponent } from './ToastComponent';
export { default as ToastComposer } from './ToastComposer';
export { default as ToastContainerComponent } from './ToastContainerComponent';
export { default as ToastPortalComposer } from './ToastPortalComposer';
export { default as ToastProvider } from './ToastProvider';
