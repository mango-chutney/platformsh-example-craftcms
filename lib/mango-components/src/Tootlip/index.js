// @flow

export { default as Tootlip } from './Tootlip';
export { default as TootlipArrowComponent } from './ArrowComponent';
export { default as TootlipPopperComponent } from './PopperComponent';
export { default as UnmanagedTootlip } from './UnmanagedTootlip';
export {
  default as UnmanagedTootlipComposer,
} from './UnmanagedTootlipComposer';
export { default as placements } from './placements';
