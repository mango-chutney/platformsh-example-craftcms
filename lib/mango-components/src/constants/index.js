import { rem } from 'polished';

export const palette = {
  primary: '#2ea1f8',
  secondary: '#1bb934',
  tertiary: '#222c3c',
  lightGray: '#fafbfc',
  mediumGray: '#cacaca',
  darkGray: '#7f8fa4',
  white: '#fefefe',
  black: '#354052',
  body: '#7f8fa4',
  heading: '#354052',
  subheading: '#7f8fa4',
  border: '#e2e7ee',
  success: '#00D161',
  alert: '#F85359',
  inactive: '#C5D0DE',
};

export const fontWeights = {
  light: 300,
  normal: 'normal',
  medium: 500,
  semibold: 600,
  bold: 800,
  extrabold: 900,
};

export const fontStack =
  'proxima-nova,Helvetica Neue Light,Helvetica Neue,Helvetica,Arial,sans-serif';

export const globalRadius = rem(4);

export const globalBorder = `${rem(1)} solid ${palette.border}`;
