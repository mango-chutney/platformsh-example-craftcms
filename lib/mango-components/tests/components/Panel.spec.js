// @flow

import 'raf/polyfill';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { Panel, PanelHeading, PanelSection } from '../../src';

function ExamplePanel() {
  return (
    <Panel>
      <PanelHeading>Panel Heading</PanelHeading>
      <PanelSection>Panel Section 1</PanelSection>
      <PanelSection>Panel Section 1</PanelSection>
    </Panel>
  );
}

it('Panel components (Panel, PanelHeading, PanelSection) renders correctly', () => {
  const tree = renderer.create(<ExamplePanel />).toJSON();
  expect(tree).toMatchSnapshot();
});
