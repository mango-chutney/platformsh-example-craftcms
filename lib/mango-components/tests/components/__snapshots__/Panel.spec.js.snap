// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel components (Panel, PanelHeading, PanelSection) renders correctly 1`] = `
.c0 {
  background: #fefefe;
  margin: 0 0 1rem;
  padding: 2rem;
  border: 0.0625rem solid #e2e7ee;
  border-radius: 0.25rem;
  overflow: hidden;
}

.c0 > .panel-section:last-child {
  margin-bottom: -2rem;
}

.c1 {
  background: #fafbfc;
  background: linear-gradient( #fafbfc 0%,#edf1f4 100% );
  border-radius: 0.25rem 0.25rem 0 0;
  border-bottom: 0.0625rem solid #e2e7ee;
  margin: -2rem -2rem 2rem;
  padding: 1rem 2rem;
  color: #354052;
  font-weight: 600;
  font-size: 1rem;
  overflow-x: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c1 + .panel-section {
  margin-top: -2rem;
}

.c2 {
  margin: 0 -2rem;
  padding: 1.5rem 2rem;
  border-bottom: 1px solid #e2e7ee;
}

.c2:first-child {
  margin-top: -2rem;
}

.c2:last-child {
  border-bottom: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Panel Heading
  </div>
  <div
    className="panel-section c2"
  >
    Panel Section 1
  </div>
  <div
    className="panel-section c2"
  >
    Panel Section 1
  </div>
</div>
`;
